osbuild/util/toml.py

```diff
@@ -0,0 +1,78 @@
+"""
+Utility functions for reading and writing toml files.
+
+Handles module imports for all supported versions (in a build root or on a host).
+"""
+import importlib
+from types import ModuleType
+from typing import Optional
+
+# Different modules require different file mode (text vs binary)
+_toml_modules = {
+    "tomllib": {"mode": "rb"},  # stdlib since 3.11 (read-only)
+    "tomli": {"mode": "rb"},  # EL9+
+    "toml": {"mode": "r", "encoding": "utf-8"},  # older unmaintained lib, needed for backwards compatibility
```

Comment:
There hasn't been a release since 2020 and Fedora deprecated it as a package in F38 (2022).
https://fedoraproject.org/wiki/Changes/DeprecatePythonToml

That's what I was going off to call it "unmaintained".

Reply:
Aha, sorry - I was looking at the wrong metric then, my bad